static/js/copy_and_paste.js

```diff
@@ -232,18 +232,17 @@ export function analyze_selection(selection) {
         }
 
         $endc = $(range.endContainer);
-        // If the selection ends in the bottom whitespace, we should
-        // act as though the selection ends on the final message.
-        // This handles the issue that Chrome seems to like selecting
-        // the compose_close button when you go off the end of the
-        // last message
-        if ($endc.attr("id") === "bottom_whitespace" || $endc.attr("id") === "compose_close") {
-            $initial_end_tr = $(".message_row").last();
-            // The selection goes off the end of the message feed, so
-            // this is a multi-message selection.
-            skip_same_td_check = true;
+        // Sometimes (especially when three click selecting in Chrome) the selection
+        // can end in the bottom whitespace, the compose close button, or an unselectable
+        // area of the next message like the unread counter. Ideally we make it so those
+        // things aren't selectable, but for now we just find the message row directly above
+        // wherever the selection ended.
+        // TODO: I'm using `undefined` to identify when $endc is of type `Text`. I imagine
+        // there's a more proper way to do this but I couldn't find one. Anyone know what to do?
+        if ($endc.prop("nodeType") !== undefined) {
+            $initial_end_tr = $endc.prev(".message_row");
         } else {
-            $initial_end_tr = $($endc.parents(".selectable_row")[0]);
+            $initial_end_tr = $($endc.parents(".message_row")[0]);
```

Comment:
I changed this so that there's a consistent search for `message_row` (following `$initial_end_tr = $(".message_row").last();` in the old code) -- thoughts?

Reply:
Can you talk about in what situation you'd expect this to be different? And also no longer setting `skip_same_td_check`

Reply:
`skip_same_td_check` is for multi-message selections, and this change is specifically to catch a case where there's only one message. I could add a separate case for `if ($endc.attr("id") === "bottom_whitespace" || $endc.attr("id") === "compose_close") {` and keep the `skip_same_td_check` -- I'm not sure what the tradeoff is between time saved and less complexity from fewer cases, but my preference was bundling them together as one case to be cleaner.

Reply:
Okay actually I think that we shouldn't `skip_same_td_check = true` here. It's not a multi-message selection just because you go off the end of a message. I'll write that in a commit message.